tests/custom-assertions.js

```diff
@@ -81,7 +139,7 @@ function assertSegments(parent, expected, options, { assert = require('node:asse
       if (typeof sequenceItem === 'string') {
         // find corresponding child in parent
         for (let j = 0; j < parent.children.length; j++) {
-          if (parent.children[j].name === sequenceItem) {
+          if (parent.children[j].name.startsWith(sequenceItem) === true) {
```

Comment:
This change is a result of the `apollo-server-express/segments.test.js` suite. Without the change, we'll get a test failure with the message:

>  AssertionError [ERR_ASSERTION]: segment "WebTransaction/Expressjs/POST//query/<anonymous>/hello" should have child "Nodejs/Middleware/Expressjs/<anonymous>"

That's because the Express tests end up with a segment tree looking like:

`WebTransaction/Expressjs/POST//query/<anonymous>/hello` => `Nodejs/Middleware/Expressjs/<anonymous>//graphql`

The non-Express tests (i.e. the ones that do not _explicitly_ load `express`) have a tree that looks like:

`WebTransaction/Expressjs/POST//query/<anonymous>/hello` => `Nodejs/Middleware/Expressjs/<anonymous>`

The `t.segments` assertion "passes" this due to a bug in its implementation.

So we'll need to update our `assertSegments` in core when we migrate this repo over.